From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct _lowcore
Patch-mainline: never, kabi
References: kabi

In 4.4.130, commit d6e925eca0f971476baf4def542580b37e0df80d (s390:
introduce execute-trampolines for branches), upstream commit
f19fbd5ed642dc31c809596412dab1ed56f2f156 switched a part of a padding to
__u16.  It made the kABI checker to complain.

Given it uses a padding, just hide it from the kABI checker.

In 4.4.130, commit bc650e567c9997293ed6421a058517e270d77e4c
(s390/alternative: use a copy of the facility bit mask), upstream commit
cf1489984641369611556bf00c48f945c77bcf02 split stfle_fac_list into two
parts of the same size.

Given the size has not changed and the bits over the first half are not
supposed to be used by anything external, just hide the change from the
kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/s390/include/asm/lowcore.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/arch/s390/include/asm/lowcore.h
+++ b/arch/s390/include/asm/lowcore.h
@@ -155,9 +155,13 @@ struct _lowcore {
 	/* Per cpu primary space access list */
 	__u32	paste[16];			/* 0x0400 */
 
+#ifdef __GENKSYMS__
+	__u8	pad_0x04c0[0x0e00-0x0440];	/* 0x0440 */
+#else
 	/* br %r1 trampoline */
 	__u16	br_r1_trampoline;		/* 0x0440 */
 	__u8	pad_0x0442[0x0e00-0x0442];	/* 0x0442 */
+#endif
 
 	/*
 	 * 0xe00 contains the address of the IPL Parameter Information
@@ -172,8 +176,12 @@ struct _lowcore {
 	__u8	pad_0x0e20[0x0f00-0x0e20];	/* 0x0e20 */
 
 	/* Extended facility list */
+#ifdef __GENKSYMS__
+	__u64	stfle_fac_list[32];		/* 0x0f00 */
+#else
 	__u64	stfle_fac_list[16];		/* 0x0f00 */
 	__u64	alt_stfle_fac_list[16];		/* 0x0f80 */
+#endif
 	__u8	pad_0x1000[0x11b0-0x1000];	/* 0x1000 */
 
 	/* Pointer to vector register save area */
